packages/integration-tests/src/npm_proxy_controller.ts

```diff
@@ -0,0 +1,62 @@
+import { execa } from 'execa';
+import path from 'path';
+import { existsSync } from 'fs';
+import fs from 'fs/promises';
+import { amplifyAtTag } from './constants.js';
+import os from 'os';
+import assert from 'assert';
+
+/**
+ * A class that orchestrates npm proxy usage in tests.
+ */
+export class NpmProxyController {
+  /**
+   * Creates NPM proxy controller.
+   */
+  constructor(private readonly workspacePath = process.cwd()) {}
+
+  setUp = async (): Promise<void> => {
+    // start a local npm proxy and publish the current codebase to the proxy
+    await execa('npm', ['run', 'clean:npm-proxy'], {
+      stdio: 'inherit',
+      cwd: this.workspacePath,
+    });
+    await execa('npm', ['run', 'vend'], {
+      stdio: 'inherit',
+      cwd: this.workspacePath,
+    });
+
+    // nuke the npx cache to ensure we are installing packages from the npm proxy
+    const { stdout } = await execa('npm', ['config', 'get', 'cache']);
+    const npxCacheLocation = path.join(stdout.toString().trim(), '_npx');
+
+    if (existsSync(npxCacheLocation)) {
+      await fs.rm(npxCacheLocation, { recursive: true });
+    }
+
+    // Force 'create-amplify' installation in npx cache by executing help command
+    // before tests run. Otherwise, installing 'create-amplify' concurrently
+    // may lead to race conditions and corrupted npx cache.
+    const output = await execa(
+      'npm',
+      ['create', amplifyAtTag, '--yes', '--', '--help'],
+      {
+        // Command must run outside of 'amplify-backend' workspace.
+        cwd: os.homedir(),
+      }
+    );
+
+    assert.match(output.stdout, /--help/);
+    assert.match(output.stdout, /--version/);
+    assert.match(output.stdout, /Show version number/);
+    assert.match(output.stdout, /--yes/);
```

Comment:
This does not create any amplify project. 

`npm create amplify@latest --yes -- --help` does two things:
1. prints help
2. installs it in npx cache (as desired side effect).

This snippet was added in `create_amplify.test.ts` long time ago and the purpose is explained in the comment. I extracted this code to avoid duplication, but otherwise it's purpose stays the same.

Reply:
I was referring to the cleanup of the `amplify` directory created in the home directory. Not a big deal though

Reply:
that doesn't seem to be happening (if it were it would be a bug).

```
bcd0740fa72c:playground sobkamil$ mkdir testapp022
bcd0740fa72c:playground sobkamil$ cd testapp022
bcd0740fa72c:testapp022 sobkamil$ npm create amplify@latest --yes -- --help
Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --yes                                               [boolean] [default: false]
bcd0740fa72c:testapp022 sobkamil$ ll
total 0
bcd0740fa72c:testapp022 sobkamil$ ll -a
total 0
drwxr-xr-x   2 sobkamil  staff   64 May  7 14:02 .
drwxr-xr-x  29 sobkamil  staff  928 May  7 14:02 ..
bcd0740fa72c:testapp022 sobkamil$
```

Reply:
oh I see, missed that it's actually not creating the local project either.